lib/commands/network.js

```diff
@@ -77,12 +77,8 @@ commands.toggleFlightMode = async function () {
 };
 
 commands.setGeoLocation = async function (location) {
-  /*
-   * TODO: Implement isRealDevice(). This method fails on
-   * real devices, it should throw a NotYetImplementedError
-   */
-  let cmd = `geo fix ${location.longitude} ${location.latitude}`;
-  return await this.adb.sendTelnetCommand(cmd);
+  let cmd = 'am startservice -e longitude ' + '"' + location.longitude + '"' + ' -e latitude ' + '"' + location.latitude + '"' + ' io.appium.settings/.LocationService';	
```

Comment:
I don't think this is how `adb.shell` works anymore. It should be taking an array. See https://github.com/appium/appium-android-driver/blob/master/lib/commands/general.js#L59-L63